psydac/ddm/cart.py

```diff
@@ -841,47 +874,50 @@ def _compute_shift_info( self, direction, disp ):
         buf_shape[direction] = m*p
 
         # Start location of send/recv subarrays
-        send_starts = np.zeros( self._ndims, dtype=int )
-        recv_starts = np.zeros( self._ndims, dtype=int )
+        send_starts          = np.zeros( self._ndims, dtype=int )
+        recv_starts          = np.zeros( self._ndims, dtype=int )
+        send_assembly_starts = np.zeros( self._ndims, dtype=int )
+        recv_assembly_starts = np.zeros( self._ndims, dtype=int )
+
         if disp > 0:
-            recv_starts[direction] = 0
-            send_starts[direction] = e-s+1
+            recv_starts[direction]          = 0
+            send_starts[direction]          = e-s+1
+            recv_assembly_starts[direction] = 0
+            send_assembly_starts[direction] = e-s+1+m*p
         elif disp < 0:
-            recv_starts[direction] = e-s+1+m*p
-            send_starts[direction] = m*p
+            recv_starts[direction]          = e-s+1+m*p
+            send_starts[direction]          = m*p
+            recv_assembly_starts[direction] = e-s+1+m*p
+            send_assembly_starts[direction] = 0
 
         # Store all information into dictionary
-        info = {'rank_dest'  : rank_dest,
-                'rank_source': rank_source,
-                'buf_shape'  : tuple(  buf_shape  ),
-                'send_starts': tuple( send_starts ),
-                'recv_starts': tuple( recv_starts )}
+        info = {'rank_dest'           : rank_dest,
+                'rank_source'         : rank_source,
+                'buf_shape'           : tuple(  buf_shape  ),
+                'send_starts'         : tuple( send_starts ),
+                'recv_starts'         : tuple( recv_starts ),
+                'send_assembly_starts': tuple( send_assembly_starts ),
+                'recv_assembly_starts': tuple( recv_assembly_starts )}
         return info
 
 #===============================================================================
-class InterfaceCartDecomposition(CartDecomposition):
+class InterfaceCartDecomposition:
     """
-    The Cartesian decomposition of an interface constructed from the Cartesian decomposition of the patches that share an interface.
-    This is built using a new inter-communicator between the cartesian grids.
+    The Cartesian decomposition of an interface constucted from the Cartesian decomposition of the patches that shares an interface.
+    This is built using a new inter-communicator between the cartition grids.
```

Comment:
Please revert these changes. You put the typos back into the docstring

Reply:
fixed in [797bf9e](https://github.com/pyccel/psydac/pull/232/commits/797bf9eb03c4f2f616a22fa7b86083b68c8809e9)